Don't make default arguments dynamic

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #118772

Noticed this while working on
https://github.com/pytorch/pytorch/issues/114590

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

ezyang committed:
Don't make default arguments dynamic

Noticed this while working on
https://github.com/pytorch/pytorch/issues/114590

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

[ghstack-poisoned]

anijain2305 commented on test/dynamo/test_unspec.py:
```diff
@@ -405,6 +405,18 @@ def fn(inputs, op_inputs_dict):
 
+    def test_defaults(self):
+        def g(x, i=8):
+            comptime.assert_static(i)
```
TIL about the usage of comptime. Really cool!

ezyang committed:
Update on "Don't make default arguments dynamic"


Noticed this while working on
https://github.com/pytorch/pytorch/issues/114590

Signed-off-by: Edward Z. Yang <ezyangmeta.com>

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

ezyang commented:
@pytorchbot merge
